### PR TITLE
Hwinata test dendrogram

### LINKED ISSUE #133: Dendrogram mode fails edge case with many clones
The dendrogram mode was tested with a simulated sample with 1000 subclones. This seems to break the node collision prevention algorithm. The resulting plot seems to not have any optimization applied. Specifiying specific `angle` or `spread` param in the dataframe does not change the result.

test tsv: `/hot/software/package/public-R-CancerEvolutionVisualization/development/hwinata-manuscript/data/multi-sample/phylogeny-1000.tsv`

code:
```
phy1000.dt <- fread(test.tsv, header = TRUE, sep = '\t');
phy1000.dt[, `:=`(
    mode = 'dendrogram',
    length.1 = 10,
    connector.type = 'solid',
    connector.col = 'black',
    connector.width = 3
    )];
phy1000.dt[parent == which(label == '1'), spread := 5];

plt <- SRCGrob(
    phy1000.dt,
    horizontal.padding = 1.5,
    yaxis1.label = 'Distance (# of SNV)',
    add.normal = TRUE,
    use.radians = FALSE,
    draw.nodes = FALSE
    );
```

Resulting plot:
Screenshot 2024-07-22 at 10 36 07 AM

### PR DESCRIPTION
**Pull request by whelena**
# Description
Edit dendrogram angle calculation such that angles are sepcified to give fixed x distances between leaf nodes.

Does not support mixed mode yet.

### Radial Mode
Screenshot 2024-07-26 at 5 08 39 PM

### Dendrogram mode
Screenshot 2024-07-26 at 5 11 46 PM

### Dendrogram with 1000 clones
Screenshot 2024-07-26 at 5 08 51 PM

# Checklist

- [x] This PR **does *NOT* contain** Protected Health Information [(PHI)](https://ohrpp.research.ucla.edu/hipaa/). A repo may ***need to be deleted*** if such data is uploaded. <br> Disclosing PHI is a ***major problem***[^1] - Even ***a small leak can be costly***[^2].
  
- [x] This PR **does *NOT* contain** germline genetic data[^3], RNA-Seq, DNA methylation, microbiome or other molecular data[^4].

[^1]: [UCLA Health reaches $7.5m settlement over 2015 breach of 4.5m patient records](https://healthitsecurity.com/news/ucla-health-reaches-7.5m-settlement-over-2015-breach-of-4.5m)
[^2]: [The average healthcare data breach costs $2.2 million, despite the majority of breaches releasing fewer than 500 records.](https://www.ponemon.org/local/upload/file/Sixth%20Annual%20Patient%20Privacy%20%26%20Data%20Security%20Report%20FINAL%206.pdf)
[^3]: [Genetic information is considered PHI.](https://www.genome.gov/about-genomics/policy-issues/Privacy#:~:text=In%202013%2C%20as%20required%20by,genetic%20information%20for%20underwriting%20purposes.)
  [Forensic assays can identify patients with as few as 21 SNPs](https://www.sciencedirect.com/science/article/pii/S1525157817305962)
[^4]: [RNA-Seq](https://www.nature.com/articles/ng.2248), [DNA methylation](https://ieeexplore.ieee.org/document/7958619), [microbiome](https://www.pnas.org/doi/pdf/10.1073/pnas.1423854112), or other molecular data can be used to predict genotypes (PHI) and reveal a patient's identity.

- [x] This PR **does *NOT* contain** other non-plain text files, such as: compressed files, images (*e.g.* `.png`, .`jpeg`), `.pdf`, `.RData`, `.xlsx`, `.doc`, `.ppt`, or other output files.

_&emsp; To automatically exclude such files using a [.gitignore](https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files) file, see [here](https://github.com/uclahs-cds/template-base/blob/main/.gitignore) for example._

- [x] I have read the [code review guidelines](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3187646/Code+Review+Guidelines) and the [code review best practice on GitHub check-list](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List).

- [x] I have set up or verified the `main` branch protection rule following the [github standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3190380/GitHub+Standards#GitHubStandards-Branchprotectionrule) before opening this pull request.

- [x] The name of the branch is meaningful and well formatted following the [standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List), using [AD_username (or 5 letters of AD if AD is too long)]-[brief_description_of_branch].
  
- [ ] I have added the major changes included in this pull request to the `CHANGELOG.md` under the next release version or unreleased, and updated the date.

### CODE REVIEW

File: R/angles.R [hunk 1]
```diff
@@ -26,43 +23,118 @@ calculate.angles.radial <- function(v, tree, spread, total.angle) {
         num.children <- length(child.ids);
 
         if (num.children > 0) {
-            parent.angle <- angles[current.node.id];
+            parent.angle <- angles[tree$tip == current.node.id];
```
Comment (dan-knight) [inline R/angles.R:26]: I assume this is to support arbitrary node IDs? (As opposed to automatically generated, ordered node IDs that can be used to index directly)

**Comment by dan-knight** [comment]
I like the improvements to angle and x-distance calculation. These utility functions are great candidates for unit testing. I'll write a few unit tests while you make the fixes on documentation, etc.

File: R/add.segs.R [hunk 2]
```diff
@@ -187,18 +226,15 @@ calculate.coords.dendrogram <- function(
...
 
     dy <- x[, length.colname];
-    x.length <- v[x$tip, 'x.length']
+    x.length <- v[x$tip, 'x.length'];
     dx <- if (is.na(x.length)) x[, 'length'] * tan(angle) else x.length;
 
     offset.x <- offset * offset.x.modifier;
-
     basex <- basex + dx + offset.x;
+    # basex <- basex + offset.x;
```
Comment (dan-knight) [inline R/add.segs.R:234]: Any reason to keep this? I don't think we should leave commented-out code here as it will probably be unclear in the future.

File: R/angles.R (hunk 1 above)
Comment (dan-knight) [inline R/angles.R:26]: Assuming you made this change to support arbitrary node IDs? (Rather than auto-generated, ordered IDs that can be indexed directly)

File: R/add.segs.R (hunk 2 above)
Comment (whelena) [reply R/add.segs.R:234]: Oops yea, I probably just missed it while cleaning up the code

File: R/angles.R (hunk 1 above)
Comment (whelena) [reply R/angles.R:26]: no, I initially change it because I thought a named list for `angles` might work better but i was worried it might get messed up since the `tree$tip` are also numeric. So the added code is actually reverting to original.

**Comment by whelena** [comment]
@dan-knight Can you double check the merge, I think the dplyr dependency was not removed properly because the 'danknight-dplyr` was too far behind the `dendrogram` branch. I tried converting it myself to remove the dplyr dependency but I'm running into other errors.

File: R/angles.R [hunk 3]
```diff
@@ -2,14 +2,11 @@ calculate.angles.radial <- function(v, tree, spread, total.angle) {
     root.node.id <- v$id[[1]];
     node.ids <- c(root.node.id);
 
+    random.scale <- median(tree$length1) * spread;
```
Comment (dan-knight) [inline R/angles.R:5]: Why is this called "random" scale?

File: DESCRIPTION [hunk 4]
```diff
@@ -1,7 +1,7 @@
 Package: CancerEvolutionVisualization
 Title: Publication Quality Phylogenetic Tree Plots
-Version: 3.0.0
-Date: 2024-05-07
+Version: 2.1.0
```
Comment (whelena) [inline DESCRIPTION:3]: why are we reverting back to 2.1?

**Review by whelena — COMMENTED** [review]
looks good to me!

File: DESCRIPTION (hunk 4 above)
Comment (dan-knight) [reply DESCRIPTION:3]: Good catch. I think Git was confused about which change took priority. I'll change it to 3.0.0.

**Comment by whelena** [comment]
I reverted and fixed it to match the code standard of the rest of it.

**Comment by whelena** [comment]
I reverted and fixed it to match the code standard of the rest of it.
